addons/website/static/src/scss/secondary_variables.scss

```diff
@@ -14,22 +14,48 @@ $o-color-palettes: append($o-color-palettes, map-merge($-palette-default, o-map-
 
 // Theme color palette
 // -------------------
+//
+// Proper themes have to enforce primary = alpha and secondary = beta.
+// Exceptions to this rule are made for compatibility reasons
+$-palette-default: nth($o-theme-color-palettes, $o-theme-color-palette-number);
+
+$-primary-default: map-get($-palette-default, 'primary');
+$-secondary-default: map-get($-palette-default, 'secondary');
+$-alpha-default: map-get($-palette-default, 'alpha');
+$-beta-default: map-get($-palette-default, 'beta');
 
-// Generate default theme color scheme if alpha is set
+// check if theme enforces primary = alpha and secondary = beta.
+$-is-alpha-primary: not $-primary-default or $-primary-default == $-alpha-default;
+$-is-beta-secondary: not $-secondary-default or $-secondary-default == $-beta-default;
+
+// Generate default theme color scheme, from primary if it's not a good theme, else from alpha
+$-primary: map-get($o-user-theme-color-palette, 'primary');
 $-alpha: map-get($o-user-theme-color-palette, 'alpha');
-@if ($-alpha) {
+
+@if ($-is-alpha-primary and $-alpha) {
     $o-user-theme-color-palette: map-merge((
+        primary: $-alpha,
+        secondary: lighten(desaturate($-alpha, 60%), 30%),
         beta: lighten(desaturate($-alpha, 60%), 30%),
         gamma: desaturate(adjust-hue($-alpha, -45deg), 10%),
         delta: desaturate(adjust-hue($-alpha, 45deg), 10%),
         epsilon: desaturate(adjust-hue($-alpha, 180deg), 10%),
     ), o-map-omit($o-user-theme-color-palette));
 }
+@else if (not $-is-alpha-primary and $-primary) {
+    $o-user-theme-color-palette: map-merge((
+        secondary: lighten(desaturate($-primary, 60%), 30%),
+        alpha: $-primary,
+        beta: lighten(desaturate($-primary, 60%), 30%),
+        gamma: desaturate(adjust-hue($-primary, -45deg), 10%),
+        delta: desaturate(adjust-hue($-primary, 45deg), 10%),
+        epsilon: desaturate(adjust-hue($-primary, 180deg), 10%),
+    ), o-map-omit($o-user-theme-color-palette));
+}
```

Comment:
Why do you need all of this? Can't we just check if primary and/or secondary is defined in the palette and if so set the `$-is-alpha-primary` / `$-is-beta-secondary` to true? We cannot just change how the color generation work in stable. People using theme clean, whether they liked it or not, have their beta/gamma/delta/epsilon computed from alpha, we can't change that without breaking every website using clean and other themes right now. This bug fix should only be about keeping the same number of color buttons, for a given theme, in the customize modal. Maybe adding a tooltip on the button that will generate the other color could be nice though.